core/embed/rust/src/ui/model_tt/component/homescreen.rs

```diff
@@ -0,0 +1,266 @@
+use crate::{
+    time::{Duration, Instant},
+    ui::{
+        component::{Component, Empty, Event, EventCtx, Pad, TimerToken},
+        display::{self, Color, Font},
+        event::{TouchEvent, USBEvent},
+        geometry::{Offset, Point, Rect},
+        model_tt::constant,
+    },
+};
+
+use super::{icon_text_center, theme, Loader, LoaderMsg, NotificationFrame};
+
+const AREA: Rect = constant::screen();
+const TOP_CENTER: Point = AREA.top_left().center(AREA.top_right());
+const LABEL_Y: i16 = 216;
+const LOCKED_Y: i16 = 101;
+const TAP_Y: i16 = 134;
+const HOLD_Y: i16 = 35;
+const LOADER_OFFSET: Offset = Offset::y(-10);
+const LOADER_DELAY: Duration = Duration::from_millis(500);
+const LOADER_DURATION: Duration = Duration::from_millis(2000);
+
+pub struct Homescreen<T> {
+    label: T,
+    notification: Option<(T, u32)>,
+    hold_to_lock: bool,
+    usb_connected: bool,
+    loader: Loader,
+    pad: Pad,
+    delay: Option<TimerToken>,
+}
+
+pub enum HomescreenMsg {
+    Dismissed,
+}
+
+impl<T> Homescreen<T>
+where
+    T: AsRef<str>,
+{
+    pub fn new(label: T, notification: Option<(T, u32)>, hold_to_lock: bool) -> Self {
+        Homescreen {
+            label,
+            notification,
+            hold_to_lock,
+            usb_connected: true,
```

Comment:
i was removing this usb_connected change detection logic previously from python impl (see https://github.com/trezor/trezor-firmware/pull/2523) because when the object is recreated, the flag might be set wrongly and no painting might occur. this seems to reintroduce that issue, but i am not sure if this works in the same way

Reply:
I saw that PR and assumed whenever homescreen is started then USB must be connected, which is probably not always the case. Tried to improve this in [631c138](https://github.com/trezor/trezor-firmware/pull/2538/commits/631c1383a82da15db631f934c43a18542624c5d2).